src/js/stores/document.js

```diff
@@ -1094,6 +1095,21 @@ define(function (require, exports, module) {
             var nextDocument = document.deleteIn(["guides", index]);
 
             this.setDocument(nextDocument, true);
+        },
+
+        /**
+         * Clears all the guides
+         *
+         * @private
+         * @param {{documentID: number}} payload
+         */
+        _handleGuidesCleared: function (payload) {
+            var documentID = payload.documentID,
+                document = this._openDocuments[documentID];
+
+            var nextDocument = document.set("guides", Immutable.List());
```

Comment:
All the functions in this store can throw if doc isn't found. I think it is safe to assume we will not intentionally send null documents to the store.